### PR TITLE
ENH: Bring 404 page from lesson example

### PR DESCRIPTION
**Pull request by jhlegarreta**
Bring 404 page from lesson example.

### CODE REVIEW

**Comment by kaitj** [comment]
Like the idea of the 404 page, but seems like it isn't loading the Carpentries template (at least in the netlify builds)

image

**Comment by jhlegarreta** [comment]
Yes, looks like it's missing some parts of the `theme`. Will have a look as time permits.

File: 404.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+layout: base
```
Comment (josephmje) [inline 404.md:2]:
```suggestion
layout: page
```

File: 404.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+root: .
+permalink: 404.html
```
Comment (josephmje) [inline 404.md:4]:
```suggestion
permalink: /404.html
```

**Review by josephmje — COMMENTED** [review]
just left a few suggestions. not sure whether they will change the rendering or not.

**Comment by jhlegarreta** [comment]
Thanks for the review @josephmje .

I tried building the site locally and the 404 HTML takes the theme correctly, so maybe it's some `netlify` artifact.
404_dmri_carpentries

**Review by kaitj — APPROVED** [review]
I am good with merging this in spite of Netlify not rendering. Did a cross-reference with the 404 page in the `SDC-BIDS-IntroMRI` repo and everything is the same. 

I suggest taking a quick look following merge to make sure the published page renders correctly

**Comment by jhlegarreta** [comment]
We went ahead and merged this hypothesizing that it was maybe a `netlify` artifact what is shown in the [picture here](https://github.com/carpentries-incubator/SDC-BIDS-dMRI/pull/112#issuecomment-810214586).

It turns out that `netlify` was correct, e.g.
https://carpentries-incubator.github.io/SDC-BIDS-dMRI/diffusion_tensor_imaging2/index.html

I've only been able to find a single relevant commit to the `404` page in the examples repository:
https://github.com/carpentries/lesson-example/commit/d8f93f065ec7d4fb24c087b5d7ff3ddec63bedac

Looks like we are missing something elsewhere. @tobyhodges do you happen to have an intuition of what could be missing? Thanks.

**Comment by jhlegarreta** [comment]
Cross-referencing #140.

**Pull request merged**